pipelines/steps/check-which-component-is-modified.yaml

```diff
@@ -16,8 +16,8 @@ steps:
     deployEverything=false
     echo "Source branch is '$sourceBranch'"
     if [[ $sourceBranch = dev ]]; then
-        targetBranch=testing
-    elif [[ $sourceBranch = testing ]]; then
+        echo "The source branch is dev, so forcing a full deployment"
+        deployEverything=true
```

Comment:
I added `deployEverything=true` here incase something else had been deployed to test, so that all new changes are deployed there. Theoretically this should be fast since only modified infra can ever be applied by Terraform.

i.e. the assumption is that dev/build envs will only be touched by this process, while test/main _could_ have a different state. The dev/build envs can be fully restored to a specific state by running the pipeline manually